### DISCUSSION TITLE
how to register a event handler with nodejs api?

### DISCUSSION THREAD

**Discussion by levscaut** [webpack/webpack#16357] [Q&A]
in normal nodejs, a simple http server can create like this:
```
var http = require('http');
var server = http.createServer();
```
and register an event handler like this:
`
server.on('request', (req, res)=>{
	// record current time in log file
})
`
so how to register an event handler with WebpackDevServer create with this?
`
const devServer = new WebpackDevServer(serverConfig, compiler);
`

I really need this solution because I'm new to web development and want to know when my website is being visited. Thanks a lot!

**Comment by alexander-akait** [comment]
You should not use webpack dev server for production, it was not design for this
